### PR TITLE
python3Packages.chispa: init at 0.8.3

### PR DESCRIPTION
**Pull request by ratsclub**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by arthurhenrique — APPROVED** [review]

**Review by giovannicaldeira — APPROVED** [review]

File: pkgs/development/python-modules/chispa/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    owner = "MrPowers";
+    rev = "v${version}";
+    sha256 = "sha256-dKiBdJNnPAJQp4OPtUjBPMqZSLNCDhfQ6OwJyHtvTrw=";
+  };
+
+  nativeBuildInputs = [ poetry ];
+
+  meta = with lib; {
+    homepage = "https://github.com/MrPowers/chispa";
+    description = "PySpark test helper methods with beautiful error messages.";
```
Comment (IvarWithoutBones) [inline pkgs/development/python-modules/chispa/default.nix:19]: There shouldn't be a dot at the end of this:
```suggestion
    description = "PySpark test helper methods with beautiful error messages";
```
Other than that, this lgtm :blush:

Comment (ratsclub) [reply pkgs/development/python-modules/chispa/default.nix]: Thanks for reviewing!

File: pkgs/development/python-modules/chispa/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{ buildPythonPackage, fetchFromGitHub, lib, poetry, ... }:
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/chispa/default.nix:1]:
```suggestion
{ buildPythonPackage, fetchFromGitHub, lib, poetry }:
```

File: pkgs/development/python-modules/chispa/default.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    repo = "chispa";
+    owner = "MrPowers";
+    rev = "v${version}";
+    sha256 = "sha256-dKiBdJNnPAJQp4OPtUjBPMqZSLNCDhfQ6OwJyHtvTrw=";
+  };
+
+  nativeBuildInputs = [ poetry ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/chispa/default.nix:15]: Can we use poetry-core?

Comment (ratsclub) [reply pkgs/development/python-modules/chispa/default.nix:15]: I get the exact problem described here: https://github.com/NixOS/nixpkgs/issues/103325

Can I keep `poetry` instead?

Comment (ratsclub) [reply pkgs/development/python-modules/chispa/default.nix:15]: Opened a PR upstream trying to fix this. 

https://github.com/MrPowers/chispa/pull/40

Comment (SuperSandro2000) [reply pkgs/development/python-modules/chispa/default.nix:15]: You can fetch the commit as a patch.

**Comment by ratsclub** [comment]
@SuperSandro2000 Got it merged upstream now. Thanks for pointing it out!

**Review by IvarWithoutBones — COMMENTED** [review]
Could we add `pythonImportsCheck`?

File: pkgs/development/python-modules/chispa/default.nix (hunk 3 above)
Comment (IvarWithoutBones) [reply pkgs/development/python-modules/chispa/default.nix]: Already got included in an upstream release, it seems

**Comment by IvarWithoutBones** [comment]
Not sure, maybe @SuperSandro2000 knows?

**Comment by ratsclub** [comment]
@IvarWithoutBones I think I got it working!

**Review by IvarWithoutBones — APPROVED** [review]
Ah, that solution makes a lot of sense. This looks good to me :blush:

Result of `nixpkgs-review pr 160387` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.chispa</li>
    <li>python39Packages.chispa</li>
  </ul>
</details>

**Comment by ratsclub** [comment]
Thank you, @IvarWithoutBones !

**Pull request merged**